packages/babel-plugin-proposal-class-static-block/src/index.js

```diff
@@ -30,30 +30,28 @@ export default declare(({ types: t, template, assertVersion }) => {
         const { scope } = path;
         const classBody = path.get("body");
         const privateNames = new Set();
-        let staticBlockPath;
-        for (const path of classBody.get("body")) {
+        const body = classBody.get("body");
+        for (const path of body) {
           if (path.isPrivate()) {
             privateNames.add(path.get("key.id").node.name);
-          } else if (path.isStaticBlock()) {
-            staticBlockPath = path;
           }
         }
-        if (!staticBlockPath) {
-          return;
+        for (const path of body) {
+          if (!path.isStaticBlock()) continue;
+          const staticBlockPrivateId = generateUid(scope, privateNames);
```

Comment:
The inserted unique staticBlockPrivateId may accidentally shadowed private id defined on upper levels:

```js
class C {
  static #_;
  constructor() {
    class D {
      static {
        C.#_ = 42;
      }
    }
  }
}
```
The injected `#_ = AIIFE(static block)` will shadow `#_` defined on `C`. Consider reuse the `privateNameVisitorFactory` in `@babel/helper-create-class-features-plugin`.

Reply:
Why don't we just use the filename + source location to generate a private name?

Reply:
We should probably consider tracking private identifiers in `@babel/traverse`'s `generateUid`.

Reply:
Or we provide a new `generateUniquePrivateKey`API since plain identifier does not conflict with private identifiers.